README
                                        mmmmm m     m mmmm     mm   m    m
                                         #   "# "m m"  #   "m   ##   "m  m"
                                         #mmm#"  "#"   #    #  #  #   #  #
                                         #        #    #    #  #mm#   "mm"
                                         #        #    #mmm"  #    #   ##
                                         Author : Justin Ball SN: nullr00t
                                         Date: 2018-10-21
                                         https://github.com/nullr00t

    pyDav is meant as an interactive script to install the dav2fs package on your system and easily mount a webDAV
 folder that you already know the URL of. I later plan to expand this script to multiple distros and package managers.
 Windows and macOS already have a semi easy way of mounting webDAV folders as network volumes, so I thought why not
 create a script that will take most of the work out of it and at least prompt you for the information. Initially I have
 started using GMX's webDAV folder for experimenting with this on my personal account. Any feedback is welcome.

 pydav.py is a fully interactive script that will walk you through the process using your system's default python
 interpreter. That being said, I have written it on a system whose default interpreter is python 3 based, therefor it is
 written in python3 syntax.

 pydav4py2.py is the same as pydav.py but for systems with python 2 as the default interpreter.

 pydavmin3.py is a barebones script for those who want to define absolute mount points themselves and require no walking
 through of how it works. requires you have dav2fs package installed on your system.

 pydavmin2.py is exactly the same as pydavmin3, just for systems with python 2 as the default interpreter.


changelog
########################################################
########## Change Log for pyDav ########################
########################################################


2018-10-14

        Intial commit to github. Code is currently NOT in a working state. That will change soon once I have time
        to work on it. Current version is 0.1, Alpha release.

2018-10-14

        Most of the code is working. Added comments to code to help with trouble spots. Updated version to 0.1.1

2018-10-15
        The script is working! Now on to improvements and making a minimalistic structured version of it!
        Any help making this version more secure or compliant is appreciated! Version 1.0!

2018-10-17
        Changed over to version 1.1 ; re wrote the script for better understanding of logic and error handling.

2018-10-21
        Version 1.2 on  pydav.py Added some style to the way it is presented in the terminal. Fixed some typos.
        Fixed an error where an exit was not possible if selected due to the way the code was executed.
        Added option to unmount.
        Version 1.0 on pydav4py2, pydavmin2, and pydavmin3.
        Removing prior pydav repos from github. Keeping changelog in tact.


pydav4py2.py
#!/usr/bin/python
import os
import getpass

print('''
        WELCOME TO

 mmmmm m     m mmmm     mm   m    m
 #   "# "m m"  #   "m   ##   "m  m"
 #mmm#"  "#"   #    #  #  #   #  # 
 #        #    #    #  #mm#   "mm" 
 #        #    #mmm"  #    #   ##  
                Version 1.0 for Python2 
                Explicitly Meant To Be Ran In Python2
    \033[1;91mThe easy python script for setting up mounting and unmounting webDAV folders.\033[1;m
    \033[1;32m+ -- -- +=[ Original Author: Justin Ball sn:nullr00t\033[1;m
    \033[1;91mCheck out my repos on https://github.com/nullr00t for other programs or scripts by me.\033[1;m
    \033[1;91mIf you already have dav2fs installed and have your folders set up,\033[1;m
    \033[1;91mand have a broader understanding of these scripts please feel free to run pydavmin2.py \033[1;m
    \033[1;91mScript was originally wrote for Kubuntu 18.04,other Debian based and apt package managed systems \033[1;m
    \033[1;91mmay work but are not guarunteed. Script depends on having the davfs2 package installed on your\033[1;m
    \033[1;91msystem as well as being able to provide a password for sudo privileges to install dav2fs.\033[1;m
    \033[1;91mI hope you find this script useful!\033[1;m
            ''')


def options():
    print('''
                1) Install the davfs2 package
                2) Mount webDAV folder
                3) Unmount webDAV folder
                4) Exit
                 ''')


while True:
        try:
            options()
            choice = raw_input('Enter your choice [ 1 - 4 ]\n')
            choice = int(choice)
            acc_strings = {'Yes', 'y', 'Y', 'yes', 'Ye', 'ye'}
            unacc_strings = {'No', 'n', 'N', 'no'}
            numinput = {1, 2, 3, 4}
            if choice == 1:
                os.system("sudo apt-get update && sudo apt-get install davfs2 -y")
            if choice == 4:
                print("Script termination requested, goodbye...")
                exit()
            elif choice == 2:
                os.system("cd ~/")
                os.system("ls -l ~/")
                print("\n\033[1;91m[Info]These are files and directories in your home folder.\033[1;m")
                print("\n\033[1;91m[Info]If the line starts with a d, then it is a directory.\033[1;m")
                choice2opt2 = raw_input("Would you like to make a new folder in your home directory?:\n")
                choice2opt2 = str(choice2opt2)
                if choice2opt2 in acc_strings:
                    newdir = raw_input("Enter the new folder's name:\n")
                    os.system("mkdir" + " " + "~/" + newdir)
                    davurl = raw_input("Please type in the URL of the webDAV folder you would like to mount:\n")
                    u = getpass.getuser()
                    os.system("sudo mount -t davfs -o uid=" + u + "," + "gid=" + u + " " + davurl + " " + "~/" + newdir)
                    print("Operation has completed, see line above for error any error info. Exiting...")
                    exit()
            elif choice == 3:
                choice3opt1 = raw_input("Please type the address of the webDAV server that you have mounted:\n")
                choice3opt1 = str(choice3opt1)
                os.system("sudo umount" + " " + choice3opt1)
                print("Operation has completed, see line above for any error information.")
            elif choice not in numinput:
                print("Not a valid option selection. Please try again.")
        except ValueError:
            print("Error 1 - Not a valid input type. Please input a number from the options menu.\n")
        except NameError:
            print("Error 2 - Not a valid input type. Please input a number from the options menu.\n")


pydavmin2.py
#!/usr/bin/python
import os
import getpass


def usage():
    print(''' 
    This script assumes that you have dav2fs installed on your system. 
    This script is meant to be run with a python2 interpreter.
    If you do not have it installed, please run sudo apt-get install davfs2, on Ubuntu and Debian based distros
    Otherwise consult your distro's package manager and search for appropriate package.
          ''')


def options():
    print('''
                1) Mount webDAV folder
                2) Unmount webDAV folder
                3) Exit
                 ''')


usage()
while True:
        try:
            options()
            choice = raw_input('Enter your choice [ 1 - 4 ]\n')
            choice = int(choice)
            acc_strings = {'Yes', 'y', 'Y', 'yes', 'Ye', 'ye'}
            unacc_strings = {'No', 'n', 'N', 'no'}
            if choice == 3:
                print("Script termination requested, goodbye...")
                exit()
            elif choice == 1:
                os.system("cd ~/")
                os.system("ls -l ~/")
                print("\n\033[1;91m[Info]These are files and directories in your home folder.\033[1;m")
                print("\n\033[1;91m[Info]If the line starts with a d, then it is a directory.\033[1;m")
                choice2opt2 = raw_input("Would you like to make a new folder in your home directory?:\n")
                choice2opt2 = str(choice2opt2)
                if choice2opt2 in acc_strings:
                    newdir = raw_input("Enter the new folder's name:\n")
                    os.system("mkdir" + " " + "~/" + newdir)
                    davurl = raw_input("Please type in the URL of the webDAV folder you would like to mount:\n")
                    u = getpass.getuser()
                    os.system("sudo mount -t davfs -o uid=" + u + "," + "gid=" + u + " " + davurl + " " + "~/" + newdir)
                    print("Operation has completed, see line above for error any error info. Exiting...")
                    exit()
            elif choice == 2:
                choice3opt1 = raw_input("Please type the address of the webDAV server that you have mounted:\n")
                choice3opt1 = str(choice3opt1)
                os.system("sudo umount" + " " + choice3opt1)
                print("Operation has completed, see line above for any error information.")
            else:
                print("Option number outside expected range. Please input a number from the options menu.\n")
        except ValueError:
            print("Not a valid input type. Please input a number from the options menu.\n")
        except NameError:
            print("Not a valid input type. Please input a number from the options menu.\n")


pydavmin3.py
#!/usr/bin/python
import os
import getpass


def usage():
    print(''' 
    This script assumes that you have dav2fs installed on your system. 
    This script is meant to be run with a python3 interpreter.
    If you do not have it installed, please run sudo apt-get install davfs2, on Ubuntu and Debian based distros
    Otherwise consult your distro's package manager and search for appropriate package.
          ''')


def options():
    print('''
                1) Mount webDAV folder
                2) Unmount webDAV folder
                3) Exit
                 ''')


usage()
while True:
        try:
            options()
            choice = input('Enter your choice [ 1 - 4 ]\n')
            choice = int(choice)
            acc_strings = {'Yes', 'y', 'Y', 'yes', 'Ye', 'ye'}
            unacc_strings = {'No', 'n', 'N', 'no'}
            if choice == 3:
                print("Script termination requested, goodbye...")
                exit()
            elif choice == 1:
                os.system("cd ~/")
                os.system("ls -l ~/")
                print("\n\033[1;91m[Info]These are files and directories in your home folder.\033[1;m")
                print("\n\033[1;91m[Info]If the line starts with a d, then it is a directory.\033[1;m")
                choice2opt2 = input("Would you like to make a new folder in your home directory?:\n")
                choice2opt2 = str(choice2opt2)
                if choice2opt2 in acc_strings:
                    newdir = input("Enter the new folder's name:\n")
                    os.system("mkdir" + " " + "~/" + newdir)
                    davurl = input("Please type in the URL of the webDAV folder you would like to mount:\n")
                    u = getpass.getuser()
                    os.system("sudo mount -t davfs -o uid=" + u + "," + "gid=" + u + " " + davurl + " " + "~/" + newdir)
                    print("Operation has completed, see line above for error any error info. Exiting...")
                    exit()
            elif choice == 2:
                choice3opt1 = input("Please type the address of the webDAV server that you have mounted:\n")
                choice3opt1 = str(choice3opt1)
                os.system("sudo umount" + " " + choice3opt1)
                print("Operation has completed, see line above for any error information.")
            else:
                print("Option number outside expected range. Please input a number from the options menu.\n")
        except ValueError:
            print("Not a valid input type. Please input a number from the options menu.\n")
        except NameError:
            print("Not a valid input type. Please input a number from the options menu.\n")


pydav.py
#!/usr/bin/python
import os
import getpass

print('''
        WELCOME TO

 mmmmm m     m mmmm     mm   m    m
 #   "# "m m"  #   "m   ##   "m  m"
 #mmm#"  "#"   #    #  #  #   #  # 
 #        #    #    #  #mm#   "mm" 
 #        #    #mmm"  #    #   ##  
                Version 1.2 for Python3 
    \033[1;91mThe easy python script for setting up mounting and unmounting webDAV folders.\033[1;m
    \033[1;32m+ -- -- +=[ Original Author: Justin Ball sn:nullr00t\033[1;m
    \033[1;91mCheck out my repos on https://github.com/nullr00t for other programs or scripts by me.\033[1;m
    \033[1;91mIf you already have dav2fs installed and have your folders set up,\033[1;m
    \033[1;91mand have a broader understanding of these scripts please feel free to run pydavmin3.py \033[1;m
    \033[1;91mScript was originally wrote for Kubuntu 18.04,other Debian based and apt package managed systems \033[1;m
    \033[1;91mmay work but are not guarunteed. Script depends on having the davfs2 package installed on your\033[1;m
    \033[1;91msystem as well as being able to provide a password for sudo privileges to install dav2fs.\033[1;m
    \033[1;91mI hope you find this script useful!\033[1;m
            ''')


def options():
    print('''
                1) Install the davfs2 package
                2) Mount webDAV folder
                3) Unmount webDAV folder
                4) Exit
                 ''')


while True:
        try:
            options()
            choice = input('Enter your choice [ 1 - 4 ]\n')
            choice = int(choice)
            acc_strings = {'Yes', 'y', 'Y', 'yes', 'Ye', 'ye'}
            unacc_strings = {'No', 'n', 'N', 'no'}
            numinput = {1, 2, 3, 4}
            if choice == 1:
                os.system("sudo apt-get update && sudo apt-get install davfs2 -y")
            if choice == 4:
                print("Script termination requested, goodbye...")
                exit()
            elif choice == 2:
                os.system("cd ~/")
                os.system("ls -l ~/")
                print("\n\033[1;91m[Info]These are files and directories in your home folder.\033[1;m")
                print("\n\033[1;91m[Info]If the line starts with a d, then it is a directory.\033[1;m")
                choice2opt2 = input("Would you like to make a new folder in your home directory?:\n")
                choice2opt2 = str(choice2opt2)
                if choice2opt2 in acc_strings:
                    newdir = input("Enter the new folder's name:\n")
                    os.system("mkdir" + " " + "~/" + newdir)
                    davurl = input("Please type in the URL of the webDAV folder you would like to mount:\n")
                    u = getpass.getuser()
                    os.system("sudo mount -t davfs -o uid=" + u + "," + "gid=" + u + " " + davurl + " " + "~/" + newdir)
                    print("Operation has completed, see line above for error any error info. Exiting...")
                    exit()
            elif choice == 3:
                choice3opt1 = input("Please type the address of the webDAV server that you have mounted:\n")
                choice3opt1 = str(choice3opt1)
                os.system("sudo umount" + " " + choice3opt1)
                print("Operation has completed, see line above for any error information.")
            elif choice not in numinput:
                print("Not a valid option selection. Please try again.")
        except ValueError:
            print("Error 1 - Not a valid input type. Please input a number from the options menu.\n")
        except NameError:
            print("Error 2 - Not a valid input type. Please input a number from the options menu.\n")
